### PR TITLE
42 implement retryexecutor with configurable retry logic

### PR DESCRIPTION
**Pull request by Abdii111**
## Summary by CodeRabbit

- **New Features**
	- Introduced a flexible, configurable retry mechanism that automatically reattempts operations on failure using adjustable attempt counts, exponential backoff delays, and optional callbacks for improved resilience.
	- Expanded the `Arrays` utility class with new methods for pairing, zipping, weaving, transposing, and chunking arrays and lists.
- **Bug Fixes**
	- Removed the `ArraysTest` class, which previously validated the `Arrays.transpose` method.
- **Tests**
	- Added a comprehensive suite of automated tests for the `Arrays` utility class, covering various functionalities and edge cases.

### CODE REVIEW

File: src/main/java/RetryExecutor.java
Comment (ArianG421) [inline src/main/java/RetryExecutor.java:1]: Need to fix the unreachable return statement at row 78.

**Review by ArianG421 — CHANGES_REQUESTED** [review]
Need to fix the unreachable return statement at row 78.

**Review by DeenoDev00 — CHANGES_REQUESTED** [review]
Need to add correct documentation.

**Review by ArianG421 — DISMISSED** [review]
Looks good, fix was made!

File: src/main/java/RetryExecutor.java
Comment (Abdii111) [reply src/main/java/RetryExecutor.java:1]: resolved

**Review by DeenoDev00 — DISMISSED** [review]
Documentation added. Approved

**Review by ArianG421 — COMMENTED** [review]
Looks good.

**Review by ArianG421 — APPROVED** [review]

**Review by DeenoDev00 — APPROVED** [review]
Nice fix, and documentation is clean. Good job! approved!

File: src/test/java/RetryExecutorTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    // Test retry logic for exception
+    @Test
+    void testExecute_RetryOnException() {
+        RetryExecutor executor = RetryExecutor.builder()
+                .maxAttempts(3)
+                .exponentialBackoff(100, 1000)
+                .retryOn(IOException.class)
+                .build();
+
+        // Vi förväntar oss att ett undantag kastas eftersom alla retries misslyckas
```
Comment (kappsegla) [inline src/test/java/RetryExecutorTest.java:31]: Keep the comments to same language. Recommend using English here.

File: src/test/java/RetryExecutorTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                })
+                .build();
+
+        assertThrows(RetryExecutor.RetryExhaustedException.class, () -> {
+            executor.execute(() -> {
+                throw new IOException("Failed");
+            });
+        });
+    }
+}
```
Comment (kappsegla) [inline src/test/java/RetryExecutorTest.java:58]: Are there more test cases we could add? 
What happens if an exception of a type we shouldn't retry on happens? Will the code stop?
What happens if we use 0 or -1 as maxAttempts argument?
Does we have any default values for these settings? Can we create a test that verifies that our code has been called the right number of times? 

Preferable if we don't have to add Mockito or any other mocking framework as a dependency for our project to verify number of calls and instead create our own test double object in test folder that counts number of calls?

File: src/test/java/RetryExecutorTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    // Test exhaustion handler
+    @Test
+    void testExecute_OnExhaustion() {
+        RetryExecutor executor = RetryExecutor.builder()
+                .maxAttempts(3)
+                .exponentialBackoff(100, 1000)
+                .retryOn(IOException.class)
+                .onExhaustion(ex -> {
+                    System.out.println("Retry exhausted");
+                    return true;
```
Comment (kappsegla) [inline src/test/java/RetryExecutorTest.java:48]: Here we define some code that should run if the maxAttempts is reached? println is not a good way to handle this in a test. What does the return true do? Can we find some other way so we can verify that onExhaustion has been run after 3 retries?
